### ISSUE TITLE
vg unrolling process

### ISSUE THREAD

**Issue by yihangs** [vgteam/vg#4103]
In original vg paper (Nature Biotech. 2018), it says that to avoid the complications introduced by cycles and inversions, vg will transform the local graph region into a directed acyclic graph (DAG), an unrolling procedure shown in Supplementary Figs.3 and 4. Is this procedure still used in vg? Is so, can we directly use this procedure, i.e. use the unrolling function F, such that we give a .vg input graph G, F(G) gives us a new DAG graph? 

Thanks!

**Comment by jeizenga** [comment]
Yes, although that particular DAG unrolling algorithm is mostly used in VG's older legacy portions (like in `vg map`'s alignment routine). You can still access it from the command line with `vg mod`, but it's in a part of the VG code base that hasn't been given much attention for maintenance or optimization.

**Comment by yihangs** [comment]
Thanks! I have 

1. To use `vg mod`, do I need to use `-b` (break-cycles) or `-f` (unfold)? If `-f` should be used, how to set the parameter of `-f` (I don't quite understand the descriptions of `-f`)?
2. Is that correct that `vg map` still has an unrolling procedure during the alignment, but other methods such as `vg giraffe` do not need this procedure anymore?

**Comment by jeizenga** [comment]
1. The relevant options are `--dagify-step`, `--dagify-to`, and `--dagify-len-max`.
2. Yes, `vg giraffe` uses haplotypes as much as possible rather than the topology of the graph, so the DAG conversion is usually unnecessary.

**Comment by faithokamoto** [comment]
Seems this was answered at the time.
